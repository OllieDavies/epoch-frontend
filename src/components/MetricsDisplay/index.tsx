import * as React from "react";
import { useEffect } from "react";
import useApi from "../useApi";
import Loading from "../Loading";

const MetricsDispaly: React.FC = () => {
  const { isLoading, response, error, fetch } = useApi("/metrics");

  // Fetch on mount and start a timer to poll
  useEffect(() => {
    fetch();
    const timer = setInterval(fetch, 3000);
    return () => clearInterval(timer);
  }, []);

  if (error) return <p>{error.response?.statusText}</p>;

  return (
    <>
      <h1>Metrics {isLoading && <Loading />}</h1>
      <code>
        <pre>{response?.data}</pre>
      </code>
    </>
  );
};
export default MetricsDispaly;
