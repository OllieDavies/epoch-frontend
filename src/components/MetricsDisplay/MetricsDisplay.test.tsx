import React from "react";
import MetricsDisplay from "./index";

describe("<MetricsDisplay />", () => {
  test("matches snapshot", () => {
    expect(<MetricsDisplay />).toMatchSnapshot();
  });
  test.todo("displays the loading component");
  test.todo("displays the api response in the code block");
  test.todo("displays http status error if response is not ok");
});
