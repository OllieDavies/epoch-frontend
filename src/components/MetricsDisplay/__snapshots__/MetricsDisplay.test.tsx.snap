// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetricsDisplay /> matches snapshot 1`] = `<MetricsDispaly />`;
